haystack/document_stores/pinecone.py

```diff
@@ -686,31 +1029,94 @@ def query_by_embedding(
         if self.similarity == "cosine":
             self.normalize_embedding(query_emb)
 
-        res = self.pinecone_indexes[index].query(query_emb.tolist(), top_k=top_k, include_values=False, filter=filters)
+        res = self.pinecone_indexes[index].query(
+            query_emb.tolist(),
+            namespace=self.embedding_namespace,
+            top_k=top_k,
+            include_values=return_embedding,
+            include_metadata=True,
+            filter=filters,
+        )
 
         score_matrix = []
         vector_id_matrix = []
+        meta_matrix = []
+        embedding_matrix = []
         for match in res["matches"]:
             score_matrix.append(match["score"])
             vector_id_matrix.append(match["id"])
-        documents = self.get_documents_by_id(vector_id_matrix, index=index, return_embedding=return_embedding)
+            meta_matrix.append(match["metadata"])
+            if return_embedding:
+                embedding_matrix.append(match["values"])
+        if return_embedding:
+            values = embedding_matrix
+        else:
+            values = None
+        documents = self._get_documents_by_meta(
+            vector_id_matrix, meta_matrix, values=values, index=index, return_embedding=return_embedding
+        )
 
         # assign query score to each document
         scores_for_vector_ids: Dict[str, float] = {str(v_id): s for v_id, s in zip(vector_id_matrix, score_matrix)}
-        for i, doc in enumerate(documents):
+        for doc in documents:
             score = scores_for_vector_ids[doc.id]
             if scale_score:
                 score = self.scale_to_unit_interval(score, self.similarity)
             doc.score = score
 
         return documents
 
-    def _attach_embedding_to_document(self, document: Document, index: str):
+    def _get_documents_by_meta(
+        self,
+        ids: List[str],
+        metadata: List[dict],
+        values: Optional[List[List[float]]] = None,
+        namespace: Optional[str] = None,
+        index: Optional[str] = None,
+        headers: Optional[Dict[str, str]] = None,
+        return_embedding: Optional[bool] = None,
+    ) -> List[Document]:
+
+        if headers:
+            raise NotImplementedError("PineconeDocumentStore does not support headers.")
+
+        if return_embedding is None:
+            return_embedding = self.return_embedding
+
+        if namespace is None:
+            if self.get_embedding_count() > 0:
+                namespace = self.embedding_namespace
+            else:
+                namespace = self.document_namespace
+
+        index = str(index or self.index)
+        index = _sanitize_index_name(index)
+
+        # extract ID, content, and metadata to create Documents
+        documents = []
+        for _id, meta in zip(ids, metadata):
+            content = meta["content"]
+            del meta["content"]
```

Comment:
:scroll: Can use `pop()` here
```suggestion
            content = meta.pop("content")
```